src/lib/client/GoodCityApiV2Client/GoodCityApiV2Client.ts

```diff
@@ -0,0 +1,24 @@
+import { ApiError } from "lib/errors";
+
+const GoodCityApiV2Client = (url: string, body: object) => {
+  return fetch(`${process.env.REACT_APP_API_V2_URL}/${url}`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  }).then(async (response) => {
+    const data = await response.json();
+    if (!response.ok) {
+      throw new ApiError({
+        httpStatus: response.status,
+        type: data.type ?? "InternalServerError",
+        message: data.error ?? "Something went wrong",
+      });
+    }
+
+    return data;
+  });
+};
+
+export default GoodCityApiV2Client;
```

Comment:
#### Few things regarding this client:

- It does a hardcoded `POST`, that'll need to be configurable for other requests
- It enforces a body
- It doesn't support custom headers
- The base url is not configurable (not a big deal)
- There's a mix of promises and async code (you can use async everywhere and avoid `then`)
- The name is a bit off (wrote another comment about that somewhere). Even though it's named `goodcity` there's nothing in there that relates to goodcity apart from the baseUrl (which could be configurable). So why not name it "json client" ?

#### Potential options

- Make it into a class, with get/post/delete/patch/put methods
- Simply make multiple functions (get/post/etc)
- Make it into an entry point which returns the methods for all actions (more functional)